### PR TITLE
[Biometrics] - KeychainError Update

### PR DESCRIPTION
**Pull request by AlejandroOrozco**
## Ticket
- [#####]

## Github Issue
- ISSUE-

## Description
- Update BiometricError with KeychainError.
- Make KeychainError public.

## Commit message
-  

## Testing
- [ ] Added unit tests
- [ ] Ran unit tests successfully
- [ ] Added documentation for public APIs and/or Wiki

### CODE REVIEW

File: TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift [hunk 1]
```diff
@@ -107,7 +107,7 @@ extension AuthenticatedSecureStorage: AuthenticatedSecureStorageProvider {
         }
 
         Logger.shared.log(withLevel: .error, message: error.localizedDescription)
-        if let biometricError = BiometricError.given(error as NSError) {
+        if let keychainError = error as? KeychainError, let biometricError = BiometricError.given(keychainError){
```
Comment (AuthyApps) [inline TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift:110]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/389d70a047903254b33925b4a9b06b5a91b8aa1c/TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift#L110"></span>Opening braces should be preceded by a single space and on the same line as the declaration.<br />
<code>opening_brace</code> <code>AuthenticatedSecureStorage.swift:110</code></td>
    </tr>
  </tbody>
</table>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift has less than 70% code coverage</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`92.76%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift | **`56.60%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift | **`92.76%`**
TwilioVerifySDK/TwilioVerify/Sources/Errors/OperationErrors.swift | **`94.23%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift has less than 70% code coverage</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`92.73%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift | **`56.60%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift | **`92.76%`**
TwilioVerifySDK/TwilioVerify/Sources/Errors/OperationErrors.swift | **`94.23%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/twilio/twilio-verify-ios/blob/9f216257ba7c7552316fcc2b2e9adbcaf448bf91/TwilioVerifySDKTests/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorageTests.swift#L24">TwilioVerifySDKTests/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorageTests.swift#L24</a> - Type body should span 200 lines or less excluding comments and whitespace: currently spans 220 lines<br />
<code>type_body_length</code> <code>AuthenticatedSecureStorageTests.swift:24</code></td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`93.43%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift | **`92.45%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift | **`93.42%`**
TwilioVerifySDK/TwilioVerify/Sources/Errors/OperationErrors.swift | **`94.23%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by SergioFierro — APPROVED** [review]

**Review by yafuquen — APPROVED** [review]

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/twilio/twilio-verify-ios/blob/9f216257ba7c7552316fcc2b2e9adbcaf448bf91/TwilioVerifySDKTests/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorageTests.swift#L24">TwilioVerifySDKTests/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorageTests.swift#L24</a> - Type body should span 200 lines or less excluding comments and whitespace: currently spans 220 lines<br />
<code>type_body_length</code> <code>AuthenticatedSecureStorageTests.swift:24</code></td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`93.40%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift | **`92.45%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift | **`93.42%`**
TwilioVerifySDK/TwilioVerify/Sources/Errors/OperationErrors.swift | **`94.23%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/twilio/twilio-verify-ios/blob/f82a02c884a051f8380030d547ac423cb0cc4772/TwilioVerifySDKTests/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorageTests.swift#L24">TwilioVerifySDKTests/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorageTests.swift#L24</a> - Type body should span 200 lines or less excluding comments and whitespace: currently spans 220 lines<br />
<code>type_body_length</code> <code>AuthenticatedSecureStorageTests.swift:24</code></td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`93.40%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Biometrics/BiometricError.swift | **`92.45%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift | **`93.42%`**
TwilioVerifySDK/TwilioVerify/Sources/Errors/OperationErrors.swift | **`94.23%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**